src/Sanderling/Sanderling.Exe/sample/script/Travel.simple.cs

```diff
@@ -1,9 +1,40 @@
-﻿//	This is a warp to 0km auto-pilot, making your travels faster and thus safer by directly warping to gates/stations.
+//	This is a warp to 0km auto-pilot, making your travels faster and thus safer by directly warping to gates/stations.
 
-while(true)
+var Measurement = Sanderling?.MemoryMeasurementParsed?.Value;
+var fly = true;
+var arriving = false;
+
+void Undock()
 {
-	var Measurement = Sanderling?.MemoryMeasurementParsed?.Value;
+	while(Measurement?.IsDocked ?? false)
+	{
+		var undockBtnText = Measurement?.WindowStation?.FirstOrDefault()?.LabelText.FirstOrDefault(candidate =>
+				candidate.Text.Contains("Undock"))?.Text;
+		if (!undockBtnText.Contains("Abort")) {
+			Sanderling.MouseClickLeft(Measurement?.WindowStation?.FirstOrDefault()?.UndockButton);
+		}
+		Host.Log("waiting for undocking to complete.");
+		Host.Delay(8000);
+		Sanderling.InvalidateMeasurement();
+		Host.Delay(1000);
+		Measurement = Sanderling?.MemoryMeasurementParsed?.Value;
+	}
+}
 
+while(fly)
+{
+	Measurement = Sanderling?.MemoryMeasurementParsed?.Value;
+	if (Measurement?.IsDocked ?? false)
+	{
+		if (arriving) {
+			fly = false;
+			Host.Log("Arrived, stopping here");
+			goto loop;
+		} else {
+			Undock();
```

Comment:
I do not understand why the control flow for `Undock` has been implemented this way:
Do we want to call `Undock` multiple times? If not, what is the reason that the call to `Undock` is contained in the loop block?